tooling-cpu-load/src/test/java/org/jboss/eap/qe/microprofile/tooling/cpu/load/HighCPUUtilsTest.java

```diff
@@ -0,0 +1,70 @@
+package org.jboss.eap.qe.microprofile.tooling.cpu.load;
+
+import java.time.Duration;
+
+import javax.management.MBeanServerConnection;
+import javax.management.ObjectName;
+import javax.management.remote.JMXConnector;
+import javax.management.remote.JMXConnectorFactory;
+import javax.management.remote.JMXServiceURL;
+
+import org.awaitility.Awaitility;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.eap.qe.microprofile.tooling.cpu.load.utils.ProcessUtils;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.arquillian.ArquillianContainerProperties;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.arquillian.ArquillianDescriptorWrapper;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.creaper.ManagementClientProvider;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+
+/**
+ * Tests tooling for HighCPUUtils. Test CPU on the system is exhausted.
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+public class HighCPUUtilsTest {
+
+    private int cores = Runtime.getRuntime().availableProcessors();
+    private int coresUnderLoad = cores > 1 ? cores - 1 : 1;
+
+    @Test
+    public void testCauseMaximumCPULoadOnProcess() throws Exception {
+        Assert.assertTrue("This test must be run on RHEL or latest Fedora machines",
+                System.getProperty("os.name").contains("Linux") && (System.getProperty("os.version").contains("el")
+                        || System.getProperty("os.version").contains("fc2")));
+
+        Process cpuLoadProcess = null;
+        try (OnlineManagementClient client = ManagementClientProvider.onlineStandalone()) {
+            HighCPUUtils.causeMaximumCPULoadOnContainer(ProcessUtils.getProcessId(client),
```

Comment:
What if I run test as a client? Is the client's process ID same as Arquillian container? In such case can we increase priority of the process of the TS so that the test execution would have enough CPU time?

Reply:
Method `ProcessUtils.getProcessId(client)` is getting pid of WF/EAP server (from server runtime model). So there is always pid of WF/EAP server, NOT surefire or client pid.

Reply:
Ah I see.